Sources/SwiftKotlinFramework/KotlinTokenizer.swift

```diff
@@ -204,22 +297,30 @@ public class KotlinTokenizer: SwiftTokenizer {
 
         let bodyStart = tokens.index(where: { $0.node === declaration.body })
 
-        if  let bodyStart = bodyStart,
-            let initExpression: ASTNode = superInitExpression ?? selfInitExpression,
-            let superIndex = tokens.index(where: { $0.node === initExpression }),
-            let endOfScopeIndex = tokens[superIndex...].index(where: { $0.kind == .endOfScope && $0.value == ")" }){
-            let keyword = superInitExpression != nil ? "super" : "this"
-            let superCallTokens = Array(tokens[superIndex...endOfScopeIndex])
-                .replacing({ $0.node === initExpression }, with: [])
-                .prefix(with: initExpression.newToken(.keyword, keyword))
-                .prefix(with: initExpression.newToken(.space, " "))
-                .prefix(with: initExpression.newToken(.symbol, ":"))
-                .suffix(with: initExpression.newToken(.space, " "))
-
-            tokens.removeSubrange((superIndex - 1)...(endOfScopeIndex + 1))
-            tokens.insert(contentsOf: superCallTokens, at: bodyStart)
+        if let bodyStart = bodyStart,
+                let initExpression: ASTNode = superInitExpression ?? selfInitExpression,
+                let superIndex = tokens.index(where: { $0.node === initExpression }) {
+            // changes here to get last ")"
+            let superStart = Array(tokens[superIndex...])
+//            if let endOfScopeIndex = lastIndexOfTokens(tokens, when: { $0.kind == .endOfScope && $0.value == ")" }) {
+            // indexOfBalancingBracket does a better job of getting the last ) when there are other brackets in arguments:
+            if let e = indexOfBalancingBracket(superStart) {
+                let endOfScopeIndex = superIndex + e
+                let keyword = superInitExpression != nil ? "super" : "this"
+                let superCallTokens = Array(tokens[superIndex...endOfScopeIndex])
+                    .replacing({ $0.node === initExpression }, with: [])
+                    .prefix(with: initExpression.newToken(.keyword, keyword))
+                    .prefix(with: initExpression.newToken(.space, " "))
+                    .prefix(with: initExpression.newToken(.symbol, ":"))
+                    .suffix(with: initExpression.newToken(.space, " "))
+
+                tokens.removeSubrange((superIndex - 1)...(endOfScopeIndex + 1))
+                tokens.insert(initExpression.newToken(.linebreak, "\n"), at:bodyStart + 1)
+                tokens.insert(contentsOf: superCallTokens, at: bodyStart)
```

Comment:
Please, can you provide an example where this code works and not the previous? I would like to add it into tests